### PR TITLE
MatchData: improve error message when refer no matched existed capture group

### PR DESCRIPTION
**Pull request by MakeNowJust**
For example, `"foo".match(/(?<g1>x)?foo/).not_nil!["g1"]` raises an exception with message `"Match group 'g1' does not exist"`, but group `"g1"` exists surely and it isn't matched on this matching. I think we can improve such an error message.

### CODE REVIEW

File: src/regex/match_data.cr [hunk 1]
```diff
@@ -157,7 +157,12 @@ class Regex
     def [](group_name : String)
       match = self[group_name]?
       unless match
-        raise KeyError.new("Match group named '#{group_name}' does not exist")
+        ret = LibPCRE.get_stringnumber(@code, group_name)
+        if ret < 0
+          raise KeyError.new("Capture group named '#{group_name}' does not exist")
+        else
+          raise KeyError.new("Capture group named '#{group_name}' is not matched")
```
Comment (RX14) [inline src/regex/match_data.cr]: "was not matched" sounds better

File: src/regex/match_data.cr [hunk 2]
```diff
@@ -315,5 +320,9 @@ class Regex
     private def raise_invalid_group_index(index)
       raise IndexError.new("Invalid capture group index: #{index}")
     end
+
+    private def raise_not_matched_group_index(index)
+      raise IndexError.new("Not matched capture group index: #{index}")
```
Comment (RX14) [inline src/regex/match_data.cr]: Maybe `"Capture group index was not matched: #{index}"`?

**Comment by asterite** [comment]
@MakeNowJust I agree with @RX14 , let's use:

- "Capture group '#{name}' was not matched" in line 164
- "Capture group #{index} was not matched" in line 325

**Comment by MakeNowJust** [comment]
@asterite Ok. I'll fix it.

**Comment by MakeNowJust** [comment]
@asterite Should is `"Invalid capture group index: #{index}"` changed?

**Comment by asterite** [comment]
@MakeNowJust I think it's fine like that

**Comment by MakeNowJust** [comment]
@asterite Thank you ❤️

File: src/regex/match_data.cr [hunk 3]
```diff
@@ -158,11 +158,8 @@ class Regex
       match = self[group_name]?
       unless match
         ret = LibPCRE.get_stringnumber(@code, group_name)
-        if ret < 0
-          raise KeyError.new("Capture group named '#{group_name}' does not exist")
-        else
-          raise KeyError.new("Capture group named '#{group_name}' is not matched")
-        end
+        status = ret < 0 ? "does not exist" : "was not matched"
+        raise KeyError.new("Capture group '#{group_name}' #{status}")
```
Comment (RX14) [inline src/regex/match_data.cr]: I think having 2 if branches here is much cleaner. Removing duplication isn't always good.

Comment (MakeNowJust) [inline src/regex/match_data.cr]: I use 80 column terminal window always, so this line is hard to read if `if` nested. It's only reason.

Comment (MakeNowJust) [inline src/regex/match_data.cr]: But it's little problem, so I fixed it.

**Pull request merged**